product/views.py

```diff
@@ -0,0 +1,133 @@
+import json
+
+from django.views import View
+from django.http import JsonResponse
+from django.db.models import Min, Max, Avg
+from collections import Counter
+
+from .models import MainCategory, SubCategory, Specific, Product, SizeType, ReleaseDate, Image, ImageType, ProductSize, Size
+from sale.models import Ask, UserAsk, Bid, Order
+
+class ProductListView(View):
+    def get(self, request):
+        tag             = request.GET.get('tag', None)
+        productCaterogy = request.GET.get('productCategory', None)
+        sizeTypes       = request.GET.get('sizeType', None)
+        size            = request.GET.getlist('size', None)
+        prices          = request.GET.getlist('prices', None)
+        releaseYears    = request.GET.getlist('releaseYear', None)
+        sort            = request.GET.get('sort', None)
+        page            = int(request.GET.get('page', 1))
+
+        response_data_set = {'Pagination' : {}, 'Filter' : {}, 'Product' : {}}
+
+        if productCaterogy:
+            response_data_set['Filter']['category'] = productCaterogy
+        if tag:
+            if '_' in tag:
+                response_data_set['Filter']['category'] = tag[:-2]
+                response_data_set['Filter']['series'] = tag[:-2:-1]
+            else:
+                response_data_set['Filter']['category'] = tag
+                response_data_set['Filter']['series'] = None
+        response_data_set['Filter']['sizetypes'] = sizeTypes
+        response_data_set['Filter']['sizes'] = size
+        response_data_set['Filter']['prices'] = prices
+        response_data_set['Filter']['releaseyears'] = releaseYears
+        response_data_set['Filter']['sort'] = sort
+
+        page_size = 20
+        page_limit = page_size * page
+        page_offset = page_limit - page_size
+
+        if sort == 'most_popular':
+            temp_res = {}
+            temp_price = {}
+            product_size_list = ProductSize.objects.select_related('product')
+            product_size_list = product_size_list.prefetch_related('ask_set')
```

Comment:
```suggestion
            product_size_list = ProductSize.objects.select_related('product').prefetch_related('ask_set')
```
chaining을 활용해 한 번에 사용하셔도 됩니다

Reply:
전반적인 로직 전체를 수정해봤습니다.